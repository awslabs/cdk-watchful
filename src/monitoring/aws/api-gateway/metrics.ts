import { Metric, Statistic } from '@aws-cdk/aws-cloudwatch';
import { Duration } from '@aws-cdk/core';
import { WatchedOperation } from '../../../api-gateway';

const enum Metrics {
  FourHundredError = '4XXError',
  FiveHundredError = '5XXError',
  CacheHitCount = 'CacheHitCount',
  CacheMissCount = 'CacheMissCount',
  Count = 'Count',
  IntegrationLatency = 'IntegrationLatency',
  Latency = 'Latency',
}

const Namespace = 'AWS/ApiGateway';

export class ApiGatewayMetricFactory {
  metricErrors(apiName: string, stage: string, op?: WatchedOperation) {
    return {
      errors4XX: this.metric(Metrics.FourHundredError, apiName, stage, op).with({ label: '4XX Errors', statistic: Statistic.SUM }),
      errors5XX: this.metric(Metrics.FiveHundredError, apiName, stage, op).with({ label: '5XX Errors', statistic: Statistic.SUM }),
    };
  }

  metricCache(apiName: string, stage: string, op?: WatchedOperation) {
    return {
      hits: this.metric(Metrics.CacheHitCount, apiName, stage, op).with({ label: 'Cache Hit', statistic: Statistic.SUM }),
      misses: this.metric(Metrics.CacheMissCount, apiName, stage, op).with({ label: 'Cache Miss', statistic: Statistic.SUM }),
    };
  }

  metricCalls(apiName: string, stage: string, op?: WatchedOperation) {
    return this.metric(Metrics.Count, apiName, stage, op).with({ label: 'Calls', statistic: Statistic.SUM });
  }

  metricIntegrationLatency(apiName: string, stage: string, op?: WatchedOperation) {
    return {
      min: this.metric(Metrics.IntegrationLatency, apiName, stage, op).with({ label: 'min', statistic: Statistic.MINIMUM }),
      avg: this.metric(Metrics.IntegrationLatency, apiName, stage, op).with({ label: 'avg', statistic: Statistic.AVERAGE }),
      p90: this.metric(Metrics.IntegrationLatency, apiName, stage, op).with({ label: 'p90', statistic: 'p90' }),
      p95: this.metric(Metrics.IntegrationLatency, apiName, stage, op).with({ label: 'p95', statistic: 'p95' }),
      p99: this.metric(Metrics.IntegrationLatency, apiName, stage, op).with({ label: 'p99', statistic: 'p99' }),
      max: this.metric(Metrics.IntegrationLatency, apiName, stage, op).with({ label: 'max', statistic: Statistic.MAXIMUM }),
    };
  }

  metricLatency(apiName: string, stage: string, op?: WatchedOperation) {
    return {
      min: this.metric(Metrics.Latency, apiName, stage, op).with({ label: 'min', statistic: Statistic.MINIMUM }),
      avg: this.metric(Metrics.Latency, apiName, stage, op).with({ label: 'avg', statistic: Statistic.AVERAGE }),
      p90: this.metric(Metrics.Latency, apiName, stage, op).with({ label: 'p90', statistic: 'p90' }),
      p95: this.metric(Metrics.Latency, apiName, stage, op).with({ label: 'p95', statistic: 'p95' }),
      p99: this.metric(Metrics.Latency, apiName, stage, op).with({ label: 'p99', statistic: 'p99' }),
      max: this.metric(Metrics.Latency, apiName, stage, op).with({ label: 'max', statistic: Statistic.MAXIMUM }),
    };
  }

  protected metric(metricName: Metrics, apiName: string, stage: string, op?: WatchedOperation) {
    return new Metric({
      metricName,
      namespace: Namespace,
      period: Duration.minutes(1),
      dimensions: {
        ApiName: apiName,
        Stage: stage,
        ...op && {
          Method: op.httpMethod,
          Resource: op.resourcePath,
        },
      },
    });
  }
}

