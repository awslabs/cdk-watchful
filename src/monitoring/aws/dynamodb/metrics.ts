import { Metric } from '@aws-cdk/aws-cloudwatch';
import { Duration } from '@aws-cdk/core';

export class DynamoDbMetricFactory {
  static readonly Namespace = 'AWS/DynamoDB';

  metricConsumedCapacityUnits(tableName: string) {
    function metric(metricName: string, label: string) {
      return new Metric({
        metricName,
        label,
        namespace: DynamoDbMetricFactory.Namespace,
        period: Duration.minutes(1),
        statistic: 'Sum',
        dimensions: {
          TableName: tableName,
        },
      });
    }

    return {
      read: metric('ConsumedReadCapacityUnits', 'Consumed (Read)'),
      write: metric('ConsumedWriteCapacityUnits', 'Consumed (Write)'),
    };
  }
}
