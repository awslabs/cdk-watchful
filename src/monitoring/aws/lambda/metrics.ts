import { Metric } from '@aws-cdk/aws-cloudwatch';
import * as cdk from '@aws-cdk/core';

export class LambdaMetricFactory {
  static readonly Namespace = 'AWS/Lambda';

  metricInvocations(functionName: string) {
    return new Metric({
      metricName: 'Invocations',
      namespace: LambdaMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Sum',
      dimensions: {
        FunctionName: functionName,
      },
    });
  }

  metricDuration(functionName: string) {
    function metric(statistic: string) {
      return new Metric({
        metricName: 'Duration',
        label: statistic,
        namespace: LambdaMetricFactory.Namespace,
        period: cdk.Duration.minutes(5),
        statistic,
        dimensions: {
          FunctionName: functionName,
        },
      });
    }

    return {
      min: metric('Minimum'),
      avg: metric('Average'),
      p50: metric('p50'),
      p90: metric('p90'),
      p99: metric('p99'),
      max: metric('Maximum'),
    };
  }

  metricErrors(functionName: string) {
    return new Metric({
      metricName: 'Errors',
      namespace: LambdaMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Sum',
      dimensions: {
        FunctionName: functionName,
      },
    });
  }

  metricThrottles(functionName: string) {
    return new Metric({
      metricName: 'Throttles',
      namespace: LambdaMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Sum',
      dimensions: {
        FunctionName: functionName,
      },
    });
  }
}
