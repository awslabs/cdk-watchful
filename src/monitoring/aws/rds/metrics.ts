import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
import * as cdk from '@aws-cdk/core';

/**
 * Metrics for RDS Aurora.
 */
export class RdsAuroraMetricFactory {
  static readonly Namespace = 'AWS/RDS';

  metricDmlThroughput(clusterIdentifier: string) {
    const dbInsertThroughputMetric = new cloudwatch.Metric({
      metricName: 'InsertThroughput',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Sum',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
    const dbUpdateThroughputMetric = new cloudwatch.Metric({
      metricName: 'UpdateThroughput',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Sum',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
    const dbSelectThroughputMetric = new cloudwatch.Metric({
      metricName: 'SelectThroughput',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Sum',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
    const dbDeleteThroughputMetric = new cloudwatch.Metric({
      metricName: 'DeleteThroughput',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Sum',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
    return {
      dbInsertThroughputMetric,
      dbUpdateThroughputMetric,
      dbSelectThroughputMetric,
      dbDeleteThroughputMetric,
    };
  }

  metricBufferCacheHitRatio(clusterIdentifier: string) {
    return new cloudwatch.Metric({
      metricName: 'BufferCacheHitRatio',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Average',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
  }

  metricDbConnections(clusterIdentifier: string) {
    return new cloudwatch.Metric({
      metricName: 'DatabaseConnections',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Average',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
  }

  metricReplicaLag(clusterIdentifier: string) {
    return new cloudwatch.Metric({
      metricName: 'AuroraReplicaLag',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Average',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
  }

  metricCpuUtilization(clusterIdentifier: string) {
    return new cloudwatch.Metric({
      metricName: 'CPUUtilization',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Average',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
  }
}
