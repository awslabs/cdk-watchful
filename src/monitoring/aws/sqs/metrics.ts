import { Metric, Statistic } from '@aws-cdk/aws-cloudwatch';
import { Duration } from '@aws-cdk/core';

const enum Metrics {
  ApproximateNumberOfMessagesVisible = 'ApproximateNumberOfMessagesVisible',
  NumberOfMessagesSent = 'NumberOfMessagesSent',
  ApproximateAgeOfOldestMessage = 'ApproximateAgeOfOldestMessage',
  SentMessageSize = 'SentMessageSize'
}

const Namespace = 'AWS/SQS';

export class SqsMetricFactory {
  metricApproximateVisibleMessages(topicName: string) {
    return this.metric(Metrics.ApproximateNumberOfMessagesVisible, topicName).with({ statistic: Statistic.MAXIMUM });
  }

  metricIncomingMessages(topicName: string) {
    return this.metric(Metrics.NumberOfMessagesSent, topicName).with({ statistic: Statistic.SUM });
  }

  metricAgeOfOldestMessageInSeconds(topicName: string) {
    return this.metric(Metrics.ApproximateAgeOfOldestMessage, topicName).with({ statistic: Statistic.MAXIMUM });
  }

  metricAverageMessageSizeInBytes(topicName: string) {
    return this.metric(Metrics.SentMessageSize, topicName).with({ statistic: Statistic.AVERAGE });
  }

  protected metric(metric: Metrics, topicName: string) {
    return new Metric({
      metricName: metric,
      namespace: Namespace,
      period: Duration.minutes(5),
      dimensions: {
        TopicName: topicName,
      },
    });
  }
}
