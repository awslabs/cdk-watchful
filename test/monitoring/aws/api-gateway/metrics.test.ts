import { ApiGatewayMetricFactory } from '../../../../src/monitoring/aws/api-gateway/metrics';

const DummyApiName = 'DummyApiName';
const DummyStage = 'DummyStage';

test('snapshot test: metricErrors', () => {
  // GIVEN
  const unitToTest = new ApiGatewayMetricFactory();

  // WHEN
  const metric = unitToTest.metricErrors(DummyApiName, DummyStage);

  // THEN
  expect(metric.count4XX.metricName).toStrictEqual('4XXError');
  expect(metric.count4XX.dimensions).toStrictEqual({ ApiName: DummyApiName, Stage: DummyStage });
  expect(metric.count5XX.metricName).toStrictEqual('5XXError');
  expect(metric.count5XX.dimensions).toStrictEqual({ ApiName: DummyApiName, Stage: DummyStage });
  expect(metric).toMatchSnapshot();
});

test('snapshot test: metricIntegrationLatency', () => {
  // GIVEN
  const unitToTest = new ApiGatewayMetricFactory();

  // WHEN
  const metric = unitToTest.metricIntegrationLatency(DummyApiName, DummyStage);

  // THEN
  Object.values(metric).forEach(eachMetric => {
    expect(eachMetric.metricName).toStrictEqual('IntegrationLatency');
    expect(eachMetric.dimensions).toStrictEqual({ ApiName: DummyApiName, Stage: DummyStage });
  });
  expect(metric).toMatchSnapshot();
});

test('snapshot test: metricLatency', () => {
  // GIVEN
  const unitToTest = new ApiGatewayMetricFactory();

  // WHEN
  const metric = unitToTest.metricLatency(DummyApiName, DummyStage);

  // THEN
  Object.values(metric).forEach(eachMetric => {
    expect(eachMetric.metricName).toStrictEqual('Latency');
    expect(eachMetric.dimensions).toStrictEqual({ ApiName: DummyApiName, Stage: DummyStage });
  });
  expect(metric).toMatchSnapshot();
});

test('snapshot test: metricCalls', () => {
  // GIVEN
  const unitToTest = new ApiGatewayMetricFactory();

  // WHEN
  const metric = unitToTest.metricCalls(DummyApiName, DummyStage);

  // THEN
  expect(metric.metricName).toStrictEqual('Count');
  expect(metric.dimensions).toStrictEqual({ ApiName: DummyApiName, Stage: DummyStage });
  expect(metric).toMatchSnapshot();
});

test('snapshot test: metricCache', () => {
  // GIVEN
  const unitToTest = new ApiGatewayMetricFactory();

  // WHEN
  const metric = unitToTest.metricCache(DummyApiName, DummyStage);

  // THEN
  expect(metric.hits.metricName).toStrictEqual('CacheHitCount');
  expect(metric.hits.dimensions).toStrictEqual({ ApiName: DummyApiName, Stage: DummyStage });
  expect(metric.misses.metricName).toStrictEqual('CacheMissCount');
  expect(metric.misses.dimensions).toStrictEqual({ ApiName: DummyApiName, Stage: DummyStage });
  expect(metric).toMatchSnapshot();
});
