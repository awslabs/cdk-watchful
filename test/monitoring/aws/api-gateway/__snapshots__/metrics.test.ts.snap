// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: metricCache 1`] = `
Object {
  "hits": Metric {
    "account": undefined,
    "color": "#2ca02c",
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "Cache Hit",
    "metricName": "CacheHitCount",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Sum",
    "unit": undefined,
  },
  "misses": Metric {
    "account": undefined,
    "color": "#d62728",
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "Cache Miss",
    "metricName": "CacheMissCount",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Sum",
    "unit": undefined,
  },
}
`;

exports[`snapshot test: metricCalls 1`] = `
Metric {
  "account": undefined,
  "color": "#1f77b4",
  "dimensions": Object {
    "ApiName": "DummyApiName",
    "Stage": "DummyStage",
  },
  "label": "Calls",
  "metricName": "Count",
  "namespace": "AWS/ApiGateway",
  "period": Duration {
    "amount": 1,
    "unit": TimeUnit {
      "inMillis": 60000,
      "isoLabel": "M",
      "label": "minutes",
    },
  },
  "region": undefined,
  "statistic": "Sum",
  "unit": undefined,
}
`;

exports[`snapshot test: metricErrors 1`] = `
Object {
  "count4XX": Metric {
    "account": undefined,
    "color": "#ff7f0e",
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "4XX Errors",
    "metricName": "4XXError",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Sum",
    "unit": undefined,
  },
  "count5XX": Metric {
    "account": undefined,
    "color": "#d62728",
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "5XX Errors",
    "metricName": "5XXError",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Sum",
    "unit": undefined,
  },
}
`;

exports[`snapshot test: metricIntegrationLatency 1`] = `
Object {
  "avg": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "avg",
    "metricName": "IntegrationLatency",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Average",
    "unit": undefined,
  },
  "max": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "max",
    "metricName": "IntegrationLatency",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Maximum",
    "unit": undefined,
  },
  "min": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "min",
    "metricName": "IntegrationLatency",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Minimum",
    "unit": undefined,
  },
  "p90": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "p90",
    "metricName": "IntegrationLatency",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "p90",
    "unit": undefined,
  },
  "p95": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "p95",
    "metricName": "IntegrationLatency",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "p95",
    "unit": undefined,
  },
  "p99": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "p99",
    "metricName": "IntegrationLatency",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "p99",
    "unit": undefined,
  },
}
`;

exports[`snapshot test: metricLatency 1`] = `
Object {
  "avg": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "avg",
    "metricName": "Latency",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Average",
    "unit": undefined,
  },
  "max": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "max",
    "metricName": "Latency",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Maximum",
    "unit": undefined,
  },
  "min": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "min",
    "metricName": "Latency",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Minimum",
    "unit": undefined,
  },
  "p90": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "p90",
    "metricName": "Latency",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "p90",
    "unit": undefined,
  },
  "p95": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "p95",
    "metricName": "Latency",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "p95",
    "unit": undefined,
  },
  "p99": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "ApiName": "DummyApiName",
      "Stage": "DummyStage",
    },
    "label": "p99",
    "metricName": "Latency",
    "namespace": "AWS/ApiGateway",
    "period": Duration {
      "amount": 1,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "p99",
    "unit": undefined,
  },
}
`;
